### DISCUSSION TITLE
Help with compiling (Win10), noob instructions

### DISCUSSION THREAD

**Discussion by VKTechnician** [collin80/SavvyCAN#315] [Q&A]
Hey all I've been trying to tackle the compile of Savvycan on Win 10 for days now. I'm not familiar with command prompt applications so I'm at a loss. I've downloaded everything that is required, QT 5.15.2, latest Savvycan continuous build, Microsoft Visual Studio 2019, Microsoft Visual Code, you name it. I'm constantly doing it wrong, can anyone provide step by step instructions on how to setup/compile  Savvycan the correct way from a noob perspective? Thank you in advance

**Comment by mickeyl** [comment]
Haven't used Windows for over a decade now, but as far as I remember, it was just calling Qt's build generator `qmake` in the source directory and then `make`. Of course that's the command-line way of doing it, no idea about doing it with an IDE.

**Reply by VKTechnician** [reply]
I have tried it from both angles. Even had 2 friends who have experience with C++ and Java and they could not help with this either

**Comment by collin80** [comment]
Well, it's been a while since I set it up myself on Windows. These days I already have it set up and usually the Windows compiles are done via a CI built with AppVeyor and they already have everything set up too. 

From what I recall, the basic steps were to install VS2019, install QT, download the SavvyCAN source, then load the project with QTCreator and compile it. That's the GUI way to do it and the way I use 99% of the time myself.

Here is the AppVeyor steps that it uses for command line compiling:

"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvars64.bat"
cd c:/projects/savvycan 
C:\Qt\5.15\msvc2019_64\bin\qmake.exe CONFIG+=RELEASE SavvyCAN.pro
nmake
cd release 
C:\Qt\5.15\msvc2019_64\bin\windeployqt.exe .\SavvyCAN.exe

SavvyCAN itself really doesn't depend on anything special. Only QT and the compiler should be necessary.

**Reply by VKTechnician** [reply]
Ok, i'll give this a try and report back.

**Reply by VKTechnician** [reply]
When trying to run the QT pro file in QT creator it continues to error. When you hit "build", I get error " C1083: Cannot open include file: 'stddef': No such file or directory". The file does exist but for some reason QT creator can't find it? If I hit "qmake" I get "Warning:  target.path is not defined : install target not created " Am I missing something? screenshots attached
![Savvycan qmake](https://user-images.githubusercontent.com/79036588/111578106-dba19080-8789-11eb-8903-f2048453ec9c.png)
![Savvycan build](https://user-images.githubusercontent.com/79036588/111578143-ebb97000-8789-11eb-8821-22db977afdab.png)

**Reply by collin80** [reply]
Well I can't say I've seen that happen before. My only real guess is: are you sure you asked Visual Studio to install the C++ tools? I don't think it defaults to installing the C++ compiler and support libraries as all the cool kids have moved on to web apps and C#. It looks like you've got the msvc2019_64 files installed for QT so that part looks good.

**Reply by VKTechnician** [reply]
Well when I open VS2019 at the bottom of the screen there is an "Output" box that has "QT VS tools" on the header, so I'm assuming it is integrated with QT as a compiler. How can I verify that it is the default compiler?
![VS20129 screenshot](https://user-images.githubusercontent.com/79036588/112539656-a39cdd80-8d87-11eb-89ca-affad52693fd.png)

**Comment by yangkkokk** [comment]
I think UI is not important, the communication should be compiled separately.

**Reply by collin80** [reply]
Well, in that case, maybe you'll be happy in a couple of months. There might be a new project on the horizon
